File: src/main/java/game/Camera/ObstructionMover.java
package game.Camera;

import game.Map.Obstruction;
import javafx.scene.Node;

public class ObstructionMover {

    public void move(Obstruction obstruction, Double offsetAndDeltaY, Double offsetAndDeltaX) {
        moveX(obstruction, offsetAndDeltaX);
        moveY(obstruction, offsetAndDeltaY);
    }

    private void moveY(Obstruction obstruction, Double offsetAndDeltaY) {
        obstruction.getNode().setTranslateY(
                obstruction.getBounds().getY() + offsetAndDeltaY);
    }

    private void moveX(Obstruction obstruction, Double offsetAndDeltaX) {
        obstruction.getNode().setTranslateX(
                obstruction.getBounds().getX() + offsetAndDeltaX);
    }


}

File: src/main/java/game/Camera/ObstructionMover.kt
package game.Camera

import javafx.geometry.Rectangle2D

class ObstructionMover {

    fun getNodeY(rectangle: Rectangle2D, offsetAndDeltaY: Double): Double {
        return rectangle.minY + offsetAndDeltaY
    }

    fun getNodeX(rectangle: Rectangle2D, offsetAndDeltaX: Double, offsetAndDeltaY: Double): Rectangle2D {
        return Rectangle2D(rectangle.minX + offsetAndDeltaX, rectangle.minY + offsetAndDeltaY, rectangle.width, rectangle.height)
    }


}
